contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -388,6 +399,9 @@ transform_from_ci_as_for_likenode(Node *node, OpExpr *op, like_ilike_info_t like
 	if (IsA(leftop, Const) || !IsA(rightop, Const) ||
 		((Const *) rightop)->constisnull)
 	{
+		/* update the collation of left and right node*/
+		linitial(op->args) = (Node *) create_collate_expr(linitial(op->args), op->inputcollid);
```

Comment:
[Question] Will we need this only when leftop is a Const node? The above `if` condition will let us in when leftop is not Const and rightop is not Const. `IsA(leftop, Const) || !IsA(rightop, Const)`

Reply:
Idea is to add CollateExpr in all cases. We will get inside this block in the following conditions:
1. leftop IS const
2. rightop is NOT const
3. rightop is const but NULL

I have added test cases for the same. Please look at case 1 and 15.